tiledb/sm/serialization/array_schema.cc

```diff
@@ -210,21 +215,36 @@ Status attribute_from_capnp(
   RETURN_NOT_OK(
       (*attribute)->set_cell_val_num(attribute_reader.getCellValNum()));
 
+  // Set nullable.
+  const bool nullable = attribute_reader.getNullable();
+  RETURN_NOT_OK((*attribute)->set_nullable(nullable));
+
   // Set the fill value.
   if (attribute_reader.hasFillValue()) {
     auto fill_value = attribute_reader.getFillValue();
-    (*attribute)
-        ->set_fill_value(fill_value.asBytes().begin(), fill_value.size());
+    if (nullable) {
+      (*attribute)
+          ->set_fill_value(
+              fill_value.asBytes().begin(),
+              fill_value.size(),
+              attribute_reader.getFillValueValidity());
+    } else {
+      (*attribute)
+          ->set_fill_value(fill_value.asBytes().begin(), fill_value.size());
+    }
   }
 
-  // Set filter pipelines
+  // Set filter pipelines.
   if (attribute_reader.hasFilterPipeline()) {
     auto filter_pipeline_reader = attribute_reader.getFilterPipeline();
     std::unique_ptr<FilterPipeline> filters;
     RETURN_NOT_OK(filter_pipeline_from_capnp(filter_pipeline_reader, &filters));
     RETURN_NOT_OK((*attribute)->set_filter_pipeline(filters.get()));
   }
 
+  // Set nullable.
+  RETURN_NOT_OK((*attribute)->set_nullable(attribute_reader.getNullable()));
```

Comment:
Needs to be protected by `if (attribute_reader.hasNullable())`

Reply:
Leaving as-is, this will default to `false`.